app/src/main/java/com/bnyro/contacts/util/SmsUtil.kt

```diff
@@ -29,40 +33,71 @@ object SmsUtil {
 
     suspend fun getSmsList(context: Context) = smsRepo.getSmsList(context)
 
-    private fun getSmsManager(context: Context): SmsManager {
-        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-            context.getSystemService(SmsManager::class.java)
+    private fun getSmsManager(subscriptionId: Int?): SmsManager {
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP_MR1 && subscriptionId != null) {
+            @Suppress("DEPRECATION")
+            SmsManager.getSmsManagerForSubscriptionId(subscriptionId)
         } else {
             @Suppress("DEPRECATION")
             SmsManager.getDefault()
         }
     }
 
-    suspend fun sendSms(context: Context, address: String, body: String): SmsData? {
+    @SuppressLint("MissingPermission")
+    @RequiresApi(Build.VERSION_CODES.LOLLIPOP_MR1)
+    fun getSubscriptions(context: Context): List<SubscriptionInfo> {
+        val subscriptionManager =
+            context.getSystemService(Context.TELEPHONY_SUBSCRIPTION_SERVICE) as SubscriptionManager?
+        return subscriptionManager!!.activeSubscriptionInfoList
+    }
+
+    suspend fun sendSms(
+        context: Context,
+        address: String,
+        body: String,
+        subscriptionId: Int?
```

Comment:
We should default this to `null` imo and only pass the argument when calling the function if it's not null